DocumentsFromSnapshotMigration/src/main/java/com/rfs/RfsMigrateDocuments.java

```diff
@@ -118,14 +118,27 @@ public static class Args {
 
         @Parameter(required = false,
         names = "--documents-per-bulk-request",
-        description = "Optional.  The number of documents to be included within each bulk request sent, default 1000")
-        int numDocsPerBulkRequest = 1000;
+        description = "Optional.  The number of documents to be included within each bulk request sent. Default no max (controlled by documents size)")
+        int numDocsPerBulkRequest = Integer.MAX_VALUE;
+
+        @Parameter(required = false,
+            names = "--documents-size-per-bulk-request",
+            description = "Optional. The maximum aggregate document size to be used in bulk requests in bytes. " +
+                "Note does not apply to single document requests. Default 1 MiB")
+        long numBytesPerBulkRequest = 1024L * 1024L;
 
         @Parameter(required = false,
             names = "--max-connections",
             description = "Optional.  The maximum number of connections to simultaneously " +
                 "used to communicate to the target, default 50")
         int maxConnections = 50;
+
+        @Parameter(required = false,
+            names = "--max-requests-per-second",
+            description = "Optional. The maximum number of requests per second to the target. Default 10")
+        double maxRequestsPerSecond = 10;
```

Comment:
Confirming that if this is less than `maxConnections`, we will potentially/likely leave connections unused.
So with the default values, we'll open 10 requests in the first second (leaving 40 connections unused) and then 10 requests in the next second (leaving at least 30 connections unused, but we may be recycling some if the first requests returned already). Does that sound right?

Reply:
I've removed this